### DISCUSSION TITLE
USB device logs

### DISCUSSION THREAD

**Discussion by barakaldes** [CrowdStrike/falconpy#169] [Q&A] [answered]
Hi guys, 
I only need the usb device events (when a user write or read files from usb and things like that). Could you tell me how can i get this information?  In Crowdstrike dashboard (USB Device Control section) i can see info about the access but in the api I can not find it.
Thanks

**Comment by jshcodes** [comment]
Hi @barakaldes!

USB device events can be consumed using the [Falcon Data Replicator (FDR)](https://github.com/CrowdStrike/FDR) project. 

(Please note: the Event Streams API will contain policy changes but not USB device events.)

More details on using Falcon Data Replicator can be found in the [CrowdStrike documentation repository](https://falcon.crowdstrike.com/documentation/9/falcon-data-replicator).

Hope this helps! :smile:

**Reply by barakaldes** [reply]
Thanks! I am reviewing the Falcon Data Replicator (Primary events, Secondary events). And i don't know where are the usb device events.  :(
what is the query to get usb device info?
¡Thank you!

**Comment by tsullivan06** [comment, marked as answer]
@barakaldes - The USB events are located in the 'Data' folder events. You'll need to decompress them to start and then you can determine what event types they are. The Events Data Dictionary in the Falcon UI (see link above) can help you identify the event types that you're looking for (search for events starting with DcUsb)

**Reply by barakaldes** [reply]
Sorry but I can't find events of type "DcUsb" in the documentation of Crowdstrike. In the [FDR documentation](https://falcon.crowdstrike.com/documentation/9/falcon-data-replicator) i dont see "DcUsb" anywhere.

**Reply by tsullivan06** [reply]
It's not in the FDR documentation - it's in the Events Data Dictionary (it's in the same location as the FDR documentation in the UI)

**Reply by barakaldes** [reply]
thanks @tsullivan06 , 
I see DcUsbConfigurationDescriptor, DcUsbDeviceBlocked, DcUsbDeviceConnected....
but only one question more. We need FDR but: Do we need insights? or only with FDR is enough?
Thanks guys

**Reply by jshcodes** [reply]
Event types are dependent on modules that would produce the event. If you don't have device control, those events may not be collected
